proposals/low-level-struct-improvements.md

```diff
@@ -394,17 +388,17 @@ ref struct RS
 }
 ```
 
-The rules for assignment also need to be adjusted to account for `ref` fields.  This design only allows for `ref` assignment of a `ref` field during object construction or when the value is known to refer to the heap. Object construction includes in the constructor of the declaring type, inside `init` accessors and inside object initializer expressions. Further the `ref` being assigned to the `ref` field in this case must have *ref-safe-to-escape* greater than the receiver of the field: 
+The rules for `ref` assignment also need to be adjusted to account for `ref` fields.  This design only allows for `ref` assignment of a `ref` field during object construction or when the value is known to refer to the heap. Object construction includes in the constructor of the declaring type, inside `init` accessors and inside object initializer expressions. Further the `ref` being assigned to the `ref` field in this case must have *ref-safe-to-escape* greater than the receiver of the field: 
 
 - Constructors: The value must be *ref-safe-to-escape* outside the constructor
 - `init` accessors:  The value limited to values that are known to refer to the heap as accessors can't have `ref` parameters
 - object initializers: The value can have any *ref-safe-to-escape* value as this will feed into the calculation of the *safe-to-escape* of the constructed object by existing rules.
 
-A `ref` field can only be assigned outside a constructor when the value is known to refer to the heap. That is allowed because it is both safe at the assignment location (meets the field assignment rules for ensuring the value being assigned has a lifetime at least as large as the receiver) as well as requires no updates to the existing method invocation rules. 
+A `ref` field can only be `ref` assigned outside a constructor when the value is known to refer to the heap. That is allowed because it is both safe at the assignment location (meets the field assignment rules for ensuring the value being assigned has a lifetime at least as large as the receiver) as well as requires no updates to the existing method invocation rules. 
```

Comment:
>when the value is known to refer to the heap [](http://example.com/codeflow?start=63&length=44)

What if the lifetime of the value is known to encompass the lifetime of the `ref struct` containing the field?

Reply:
True it could be expressed that way. A `ref struct` really only has two lifetime states: usable within the current method, returnable to the containing method. If we make it such that the field lifetime has to be at least the containing `ref struct` then that would also allow for a bit more flexibility. 

I need to think about this a bit though and make sure I'm not missing a subtle rule here.